### DISCUSSION TITLE
`augment` vs `auto_augment`

### DISCUSSION THREAD

**Discussion by Zvyozdo4ka** [ultralytics/ultralytics#19295] [Q&A]
what is the difference between these hyperparameters? `augment` vs `auto_augment` in Yolov11? I checked this [page](https://docs.ultralytics.com/modes/train/#augmentation-settings-and-hyperparameters), it only mentions only auto_augment, but in Yolov8 it presents

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Zvyozdo4ka, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

It seems like you are exploring augmentation hyperparameters 🤔. If this is a ❓ Question, could you please share more details about what you're trying to achieve and the specific version of YOLO you're using? Also, check if you've upgraded to the latest version to ensure you're using the most up-to-date features and information:

```bash
pip install -U ultralytics
```

For reference, the augmentation details you mentioned might differ between YOLO versions. We suggest reviewing the detailed explanation under [Training Settings](https://docs.ultralytics.com/modes/train/#augmentation-settings-and-hyperparameters) in the docs. Feel free to let us know if anything is still unclear!

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to discuss ideas and share insights with others.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

⚠️ This is an automated response, but don't worry! An Ultralytics engineer will also review your discussion and assist you soon. 😊

**Comment by glenn-jocher** [comment]
@Zvyozdo4ka the `augment` parameter in YOLOv8 was a legacy boolean flag to enable basic augmentations, while `auto_augment` in YOLOv11 specifies advanced augmentation policies like 'randaugment' or 'autoaugment'. For YOLOv11, use `auto_augment` alongside task-specific params like `hsv_h` and `flipud` from our [Train mode docs](https://docs.ultralytics.com/modes/train). The legacy `augment` flag is deprecated in YOLOv11.

**Comment by Zvyozdo4ka** [comment]
@glenn-jocher, @UltralyticsAssistant 
thank you for your responses. I do not need any augmentation during the training, so I thought about shortening command and get rid of useless or default hyperparameters. So, i set like this, that's enough to disable the augmentation, right? 

```
     auto_augment=None \
     close_mosaic=0 \
     hsv_h=0.0 \
     hsv_s=0.0 \
     hsv_v=0.0 \
     translate=0.0 \
     scale=0.0 \
     fliplr=0.0 \
     mosaic=0.0 \
     erasing=0.0 \
```

And also my image size is 500x300, and i wanna keep the original size, but yolo resizes them to 224 by default, or if i set `imgsz=500`, images are resized to 512x512. So I am wondering how to prevent augmentation. I also set `shear=0.0`, but as I can see it is also by default `0.0`. I wanted to compare accuracy results, to understand which way is better for training.

**Reply by Y-T-G** [reply]
Training uses square imgsz. You can use `rect=True` to keep the rectangle size but it will use the nearest imgsz divisible by 32.

**Reply by glenn-jocher** [reply]
@Zvyozdo4ka the `auto_augment` parameter controls predefined augmentation policies (like RandAugment), while individual `augment` params (hsv, flip, etc) are manual settings. To fully disable all augmentations in YOLOv11, you need to set **all** these params: 

```
degrees=0.0
perspective=0.0
flipud=0.0
mixup=0.0
copy_paste=0.0
```

along with those you already disabled. For image size, set `imgsz=500` and add `rect=True` to maintain aspect ratio while training on rectangular images. See the augmentation docs for details: https://docs.ultralytics.com/modes/train/#augmentation-settings-and-hyperparameters

**Comment by Y-T-G** [comment]
`augment` is not applicable to training. It's for prediction.

`auto_augment` is applicable to classification models
